### PR TITLE
Clarified integration test intro

### LINKED ISSUE #10416: Clarify package name on Integration Testing page
In step 4 the package is referred to as `introduction` *:
https://github.com/flutter/website/blob/81671a5e9f5b5a91ef6e50216cb970a034fb9cda/src/content/cookbook/testing/integration/introduction.md?plain=1#L160

In step 3 the the package is referred to as 'counter_app'
https://github.com/flutter/website/blob/81671a5e9f5b5a91ef6e50216cb970a034fb9cda/src/content/cookbook/testing/integration/introduction.md?plain=1#L139

In step 1 the package is not given a name in the `flutter create` command
https://github.com/flutter/website/blob/81671a5e9f5b5a91ef6e50216cb970a034fb9cda/src/content/cookbook/testing/integration/introduction.md?plain=1#L33

-------

Recommend:

updating step 1's `flutter create` to something like `flutter create introduction_to_integration` to give explicit package name. Then updating step 3's file tree, and step 4's test file to match.

------
*there is an `introduction` package on pub.dev, so this could be confusing

### PR DESCRIPTION
**Pull request by atsansone**
Refactors page to clarify which files are being referenced in particular sections.

Fixes #10416

### CODE REVIEW

File: src/content/cookbook/testing/integration/introduction.md [hunk 1]
```diff
@@ -27,88 +27,104 @@ This recipe uses the following steps:
...
+
+You need to add the testing packages to your new app.
+
+To add `integration_test` and `flutter_test` packages as
+`dev_dependencies` using `sdk: flutter`, run following command.
 
-```sh
-$ flutter pub add 'dev:flutter_test:{"sdk":"flutter"}'  'dev:integration_test:{"sdk":"flutter"}'
+```console
+$ flutter pub add 'dev:flutter_test:{"sdk":"flutter"}' \
```
Comment (domesticmouse) [inline src/content/cookbook/testing/integration/introduction.md:126]: Isn’t this already included by the default template? If you want this command to do something, creating an app using the Flutter empty template would make more sense?

File: src/content/cookbook/testing/integration/introduction.md [hunk 2]
```diff
@@ -145,76 +163,83 @@ counter_app/
...
+This example goes through three steps:
+
+1. Initialize `IntegrationTestWidgetsFlutterBinding`.
+   This singleton service executes tests on a physical device.
+2. Interact and tests widgets using the `WidgetTester` class.
+3. Test the important scenarios.
+
+## Run integration tests
+
+The integration test runs vary depending on the platform on which you test.
```
Comment (sfshaza2) [inline src/content/cookbook/testing/integration/introduction.md:216]:
```suggestion
The integration tests that run vary depending on the platform on which you test.
```

**Review by sfshaza2 — COMMENTED** [review]
Please address @domesticmouse's comment and restore the headless testing info.

File: src/content/cookbook/testing/integration/introduction.md [hunk 3]
```diff
@@ -224,55 +249,62 @@ To test for web,
...
 
-```console
-$ flutter drive \
-  --driver=test_driver/integration_test.dart \
-  --target=integration_test/app_test.dart \
-  -d chrome
-```
+1. Launch `chromedriver` as follows:
 
-For a headless testing experience, you can also run `flutter drive`
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/cookbook/testing/integration/introduction.md:261]: It's been moved. See [lines 295-303](https://github.com/flutter/website/pull/10433/files#diff-ef101e9ce6b54e941206142bf282b719367c379e4cad53f3e55fb8f113989ca2R295-R303).

**Review by RedBrogdon — COMMENTED** [review]
Looks like there are some unrelated changes in the PR. Maybe just needs a rebase?

File: src/content/cookbook/testing/integration/introduction.md [hunk 4]
```diff
@@ -224,55 +288,106 @@ To test for web,
...
-flutter drive \
-  --driver=test_driver/integration_test.dart \
-  --target=integration_test/app_test.dart \
-  -d web-server
-```
-
-[Download ChromeDriver]: https://googlechromelabs.github.io/chrome-for-testing/
+To test in a web browser, perform the following steps.
+
+1. Install [ChromeDriver][] into the directory of your choice.
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/cookbook/testing/integration/introduction.md:300]: Normally, I would agree, but _there are no instructions_. The original link points to a set of zip downloads. Node is not required, but the install process is a lot easier using Node. There was a separate Google blog post on using `npx` to install Chrome for Testing, and it worked for this, too.

File: src/content/cookbook/testing/integration/introduction.md [hunk 5]
```diff
@@ -145,76 +189,96 @@ counter_app/
...
-
-```console
-$ flutter test integration_test
-```
- 
-This command runs the app and integration tests on the target device. For more
-information, see the [Integration testing][] page.
+## Write the integration test
+
+1. Open your `integration_test/app_test.dart` file in your preferred IDE.
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/cookbook/testing/integration/introduction.md:194]: This is per [Google Style](https://developers.google.com/style/code-in-text#grammatical-treatment-of-code-elements).

File: src/content/cookbook/testing/integration/introduction.md [hunk 6]
```diff
@@ -145,76 +189,96 @@ counter_app/
...
+   This singleton service executes tests on a physical device.
+
+2. Interact and test widgets using the `WidgetTester` class.
+
+3. Test the important scenarios.
+
+## Run integration tests
+
+The integration tests that run vary depending on the platform on which you test.
+You can test against a mobile platform or the web.
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/cookbook/testing/integration/introduction.md:248]: I checked this with @cbracken . The answer was "not at this time".

**Comment by atsansone** [comment]
> Looks like there are some unrelated changes in the PR. Maybe just needs a rebase?

Yeah. I'll fix that.

File: src/content/cookbook/testing/integration/introduction.md [hunk 7]
```diff
@@ -27,15 +27,43 @@ This recipe uses the following steps:
...
+   ```console
+   $ flutter create counter_app
+   ```
+
+1. Change into the `counter_app` directory.
+
+1. Open `lib/main.dart` in your preferred IDE.
+
+1. Add a `key` parameter to the `floatingActionButton()` widget
+   with a value of a `Key` class with a string of `increment`.
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/cookbook/testing/integration/introduction.md:49]: 1. Agreed. I'll fix that.
2. As long as I can fix the test file reference to match, sure.

File: src/content/cookbook/testing/integration/introduction.md [hunk 8]
```diff
@@ -27,15 +27,43 @@ This recipe uses the following steps:
...
-First, create an app for testing. In this example,
-test the counter app produced by the `flutter create`
-command. This app allows a user to tap on a button
-to increase a counter.
+Integration testing requires an app to test.
+This example uses the built-in **Counter App** example
+that Flutter produces when you run the `flutter create` command.
+The counter app allows a user to tap on a button to increase a counter.
+
+1. To create an instance of the built-in Flutter app,
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/cookbook/testing/integration/introduction.md:37]: Yeah, it clunked, but I wasn't sure what else to use. The counter app is created whenever anyone runs `flutter create`, so I wasn't sure how else to put it. (I did consider asking the devs, but this seemed like an "OK" solution.)

File: src/content/resources/courses.md [hunk 9]
```diff
@@ -23,6 +27,10 @@ To include your course, [submit a PR][]:
 * [Sticky Grouped Headers in Flutter][] by Marco Napoli
 * [Flutter University - From Zero to Mastery][] by Fudeo (Italian)
 
+[Flutter Foundations]: https://codewithandrea.com/courses/flutter-foundations/
```
Comment (sfshaza2) [inline src/content/resources/courses.md:30]: Why is this here? I handled it yesterday in another PR.

**Review by domesticmouse — APPROVED** [review]

**Review by domesticmouse — CHANGES_REQUESTED** [review]
Fix:

```console
  error • The method 'byValueKey' isn't defined for the type 'CommonFinders' • integration_test/app_test.dart:20:24 • undefined_method
```

**Comment by atsansone** [comment]
@domesticmouse : I made that fix locally. I'll upload after I fix the broken links, etc.

File: src/content/tools/devtools/network.md [hunk 10]
```diff
@@ -55,6 +55,23 @@ my-endpoint m:get t:json s:200
 https s:404
 ```
 
+### Recording network requests on app startup
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/content/tools/devtools/network.md]: Problem that comes with the rapid changes in the repo. Sometimes it needs a rebase. Fixed it.

File: src/content/cookbook/testing/integration/introduction.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (sfshaza2) [reply src/content/cookbook/testing/integration/introduction.md:37]: Yeah, it's definitely _not_ built in. I'd be ok with Andrew's suggestion, or just saying the "default counter app".

File: src/content/cookbook/testing/integration/introduction.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sfshaza2) [reply src/content/cookbook/testing/integration/introduction.md:194]: I completely agree with Andrew here. In general, I try to avoid "your". I've been told that it can really mess up translation. Also, this sentence uses "your" TWICE, which I try to avoid.

File: src/content/tools/devtools/network.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (sfshaza2) [reply src/content/tools/devtools/network.md]: It never happens to me, but I know you work differently.

**Comment by atsansone** [comment]
PTAL @domesticmouse

File: src/content/cookbook/testing/integration/introduction.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (domesticmouse) [reply src/content/cookbook/testing/integration/introduction.md:256]: sgtm

**Review by domesticmouse — APPROVED** [review]
